test/jobs/maintenance_tasks/task_job_test.rb

```diff
@@ -110,6 +110,18 @@ class TaskJobTest < ActiveJob::TestCase
       assert_equal 1, @run.reload.tick_count
     end
 
+    test ".perform_now respects status_reload_frequency for reloading status each iteration" do
+      original_frequency = MaintenanceTasks.status_reload_frequency
+      MaintenanceTasks.status_reload_frequency = 1.second
+
+      @run.expects(:reload_status).at_least_once
```

Comment:
I was hoping to have a test that could verify that we _only_ reload the status if `status_reload_frequency` has elapsed, but I couldn't find a way to simulate travelling forward in time with this test. :(